### PR TITLE
Added movie interpolation + test function

### PR DESCRIPTION
**Pull request by Henry-Best-01**
Added movie interpolation which acts linearly in time. Movie is expected of shape (t, x, y) where the x and y coordinates are generic spatial coordinates. The t value defines the snapshot number. I attempted to keep documentation like the other functions but it still may need tweaking. The time axis must correspond to the Orig_timestamps array, where the simulated times were stored. New_timestamps may be arbitrary. Extrapolation beyond the original bounds is done simply by repeating the boundary values for every pixel.

Also added is a test function which uses the sinusoidal variability as well as generic values to represent "pixel" snapshots at 3 original time steps. The test function interpolates between these values for each pixel at the following important test areas: An endpoint, a midpoint between two points, a central value which is also an original point, and an arbitrary point between an extra long gap (interpolates to 1.5 from original time values 1.0 and np.pi). The test explicitly compares the array created through my function and the expectation calculated pixel by pixel through linear interpolation. I kept it as small as I could.

After downloading the required testing python packages, I received confirmation that it succeeds. I made some plots of it working on significantly larger movies as well (1000 x 400 x 400, summed over pixels in the source plane), which I can share if you want to see it as well.

### CODE REVIEW

File: slsim/Sources/source_variability/variability.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import numpy as np
+import scipy
 
 """This class aims to have realistic variability models for AGN and Supernovae."""
```
Comment (ajshajib) [inline slsim/Sources/source_variability/variability.py:4]:
```suggestion
"""This class aims to have realistic variability models for AGN and supernovae."""
```

Comment (ajshajib) [reply slsim/Sources/source_variability/variability.py:4]: Making an insignificant commit to trigger Pre-commit QA actions again.

**Review by nkhadka21 — CHANGES_REQUESTED** [review]

File: slsim/Sources/source_variability/variability.py [hunk 2]
```diff
@@ -42,3 +43,22 @@ def sinusoidal_variability(t, amp, freq):
     :return: variability for the given time
     """
     return amp * np.sin(2 * np.pi * freq * t)
+
+def interpolate_variability(Movie, Orig_timestamps, New_timestamps):
+    """Interpolates a variable source to any given time series
```
[the remark this answers is not in the document]
Comment (nkhadka21) [reply slsim/Sources/source_variability/variability.py:48]: Also, check why pre-commit is failing and fix them.

[the remark this answers is not in the document]
Comment (Henry-Best-01) [reply slsim/Sources/source_variability/variability.py]: It looks like pre-commit was failing because the lines were too long and I had superfluous spacings to align the comments. 

Now, when I run "pre-commit run --show-diff-on-failure --color=always --all-files" locally, I receive all 5 passes, but on Github I'm receiving an error for the hook "black" failing: 
ruff.....................................................................Passed
nbqa-ruff................................................................Passed
black....................................................................Failed
- hook id: black
- files were modified by this hook

reformatted slsim/Sources/source_variability/variability.py
reformatted tests/test_Source/test_source_variability.py

All done! ✨ 🍰 ✨
2 files reformatted, 54 files left unchanged.

black-jupyter............................................................Passed
docformatter.............................................................Passed

 As for why the function needs to be here, I guess it's because it is applicable to any series of screen shots, whether it be AGN or supernovae. Is there a variable class I should put it so it may be accessed cleaner?

**Review by ajshajib — CHANGES_REQUESTED** [review]
@Henry-Best-01 Thanks for the PR! The pre-commit QA is failing for some reason. Can you kindly run these commands in your local `slsim` directory to get rid of the formatting issues and then make a commit?

```
pip install black
pip install docformatter
black .
docformatter ./* -r --black --in-place
```

**Review by ajshajib — APPROVED** [review]

File: slsim/Sources/source_variability/variability.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (nkhadka21) [reply slsim/Sources/source_variability/variability.py]: Hi @Henry-Best-01 , I am hoping that you have a plan to implement your quasar variability model.

[the remark this answers is not in the document]
Comment (Henry-Best-01) [reply slsim/Sources/source_variability/variability.py]: Yes I do! It would be silly for me to only contribute only a few lines of code.

I will open a new pull request when I have all the pieces of my black box in place.

File: tests/test_Source/test_source_variability.py [hunk 3]
```diff
@@ -26,6 +27,34 @@ def test_variability_at_t_sinusoidal(self):
         expected_result = np.array([-0.43030122, -0.97536797, -0.14773276])
         npt.assert_almost_equal(result, expected_result, decimal=5)
 
+    def test_interpolation_for_sinusoidal(self):
+        kwargs_model00 = {"amp": 1.0, "freq": 0.5}
```
Comment (sibirrer) [inline tests/test_Source/test_source_variability.py:31]: can you also describe what you are testing here? This might help the reader and assess the completeness of the test @Henry-Best-01

File: slsim/Sources/source_variability/variability.py [hunk 4]
```diff
@@ -42,3 +43,29 @@ def sinusoidal_variability(t, amp, freq):
     :return: variability for the given time
     """
     return amp * np.sin(2 * np.pi * freq * t)
+
+
+def interpolate_variability(Image_series, Orig_timestamps, New_timestamps):
```
Comment (sibirrer) [inline slsim/Sources/source_variability/variability.py:48]: input variables have conventions of lower_case. Capital letters are reserved for Classes

File: slsim/Sources/source_variability/variability.py [hunk 5]
```diff
@@ -42,3 +43,29 @@ def sinusoidal_variability(t, amp, freq):
...
+    """Interpolates a variable source to any given time series.
+
+    :param Image_series: 3 dimensional array of shape (t, x, y) relating to snapshots of
+        a variable object in source plane
+    :param Orig_timestamps: Series of timestamps which represent original snapshots
```
Comment (sibirrer) [inline slsim/Sources/source_variability/variability.py:53]: It's unclear to me how this input differs from the first axis t in the Image_series. In general, having an explicit definition of these inputs and describing what is happening would be useful. It's unclear to me when reading the docstrings what this function does

**Review by sibirrer — CHANGES_REQUESTED** [review]
Thank you @Henry-Best-01 . I have some requests on the documentation (incl. documentation/description of the tests). Overall, this is a very useful feature. I am just a bit pedantic such that it becomes useful and someone else can also build on it :)
Also, the function/feature you added is not yet accessible through the Variability class. This is ok for now, but when we want to simulate time series, then this needs to be (perhaps for @nkhadka21 )

File: slsim/Sources/source_variability/variability.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Henry-Best-01) [reply slsim/Sources/source_variability/variability.py]: Added a sentence that this will be important for future microlensing implimentation

File: tests/test_Source/test_source_variability.py (hunk 3 above)
Comment (Henry-Best-01) [reply tests/test_Source/test_source_variability.py]: I added details to my tests (all moved into test_param_until)

File: slsim/Sources/source_variability/variability.py (hunk 4 above)
Comment (Henry-Best-01) [reply slsim/Sources/source_variability/variability.py]: fixed

File: slsim/Sources/source_variability/variability.py (hunk 5 above)
Comment (Henry-Best-01) [reply slsim/Sources/source_variability/variability.py]: Made more clear

File: slsim/Util/param_util.py [hunk 6]
```diff
@@ -85,3 +86,60 @@ def magnitude_to_amplitude(magnitude, mag_zero_point):
...
+    return np.reshape(image_series, (initial_shape[0], number_of_pixels))
+
+
+def pixels_to_images(pixels, original_shape):
+    """Converts a string of pixel snapshots back into a series of image snapshots. This
+    is the inverse of images_to_pixels.
+
+    :param pixels: Series of pixel snapshots to arrange back into the original image
+        shape
+    :param original_shape: The original shape of the image snapshots
```
Comment (sibirrer) [inline slsim/Util/param_util.py:108]: I guess this is a tuple? Perhaps specify

**Review by sibirrer — APPROVED** [review]
I have one minor comment. I approve it and let @nkhadka21 make the call

File: slsim/Util/param_util.py (hunk 6 above)
Comment (Henry-Best-01) [reply slsim/Util/param_util.py]: added clarification

**Review by nkhadka21 — APPROVED** [review]
